### DISCUSSION TITLE
關於 set_option/toggle 的使用，能否使用在 options 的輪替上

### DISCUSSION THREAD

**Discussion by markwu** [rime/home#1739] [General]
目前在 bopomofo 拼音中，有三個 option

```
  - options:
      - zh_hant
      - zh_hans
      - zh_hant_tw
    states:
      - 傳統漢字
      - 简化字
      - 臺灣字形
```

我能否像是使用 toggle 去切換全形和半形一樣，讓 set_option/toggle 在這三個 option 中切換呢？例如：

```
    - { when: always, accept: "Shift+space", set_option: [zh_hant, zh_hans, zh_hant_tw]}
or
    - { when: always, accept: "Shift+space", toggle: [zh_hant, zh_hans, zh_hant_tw]}
```

或是有其他方法可以達到這樣的目的嗎？

**Comment by lotem** [comment]
寫 `toggle` 即可。

**Reply by markwu** [reply]
可是應該怎麼寫哪？找了很久文件沒找到。像是全形半形，我還有 full_shape 這個名稱，所以可以寫成

```
    - { when: always, accept: "Shift+space", toggle: full_shape}
```

`[zh_hant, zh_hans, zh_hant_tw]` 如果要在這三個 option 中輪替， toggle 後方應該怎麼寫呢？

**Reply by markwu** [reply]
我試出來了，只要寫其中一個就可以了。

```
    - { when: always, accept: "Shift+space", toggle: zh_hant}
```

然後，他就會自動在那三個 option 中輪替。謝謝回覆。

**Reply by lotem** [reply]
> 我試出來了，只要寫其中一個就可以了。
> 
> ```
>     - { when: always, accept: "Shift+space", toggle: zh_hant}
> ```
> 
> 然後，他就會自動在那三個 option 中輪替。謝謝回覆。

細看了程序，確實是這樣。
